File: ServeSys/ServeSys.API/Modules/Order/Entities/MenuItem.cs
namespace ServeSys.API.Modules.Order.Entities;

public class MenuItem
{
    public int Id { get; set; }

    public string Name { get; set; } = null!;

    public string? Description { get; set; }

    /// <summary>Giá tại thời điểm hiện tại (snapshot được lưu vào OrderItem khi đặt món)</summary>
    public decimal Price { get; set; }

    public string? ImageUrl { get; set; }

    public int DisplayOrder { get; set; }

    public bool IsAvailable { get; set; } = true;

    public DateTime CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }

    // Foreign key
    public int MenuCategoryId { get; set; }
    public MenuCategory MenuCategory { get; set; } = null!;
}

File: ServeSys.Mobile/servesys/lib/features/menu/domain/entities/menu_item.dart
class MenuItem {
  int id;
  String name;
  String? description;
  double price;
  String? imageUrl;
  int displayOrder;
  bool isAvailable;
  bool isBestSeller;

  MenuItem({
    required this.id,
    required this.name,
    this.description,
    required this.price,
    this.imageUrl,
    required this.displayOrder,
    required this.isAvailable,
    required this.isBestSeller
  });
}
